### ISSUE TITLE
Implement category sorting for styleguide

### ISSUE THREAD

**Issue by Wolfr** [Wolfr/sveltekit-jui#5]
I would like to be able to set the order of categories in an array, and for that order to be used in the styleguide.

**Comment by PaxPax** [comment]
Are you talking about something like?
```javascript
  let categories = ['basics', 'components', 'objects', 'dev-notes']

  function sortByCategories(array, data) {
    let sortedCategories = []
    array.forEach((value) => {
      data.forEach((category) => {
        if (value === category["category"]) {
          sortedCategories.push(category)
        }
      })
    })
    return sortedCategories;
  }
```

**Comment by Wolfr** [comment]
Yes! Currently the data is here but is also ran through a reduce function to get the categories out of it (by grouping them).

https://github.com/Wolfr/sveltekit-jui/blob/main/src/routes/styleguide/$layout.svelte#L21-L187

**Comment by PaxPax** [comment]
> 
> 
> Yes! Currently the data is here but is also ran through a reduce function to get the categories out of it (by grouping them).
> 
> https://github.com/Wolfr/sveltekit-jui/blob/main/src/routes/styleguide/$layout.svelte#L21-L187

Nice, we could just run the sortByCategories after the reduce and it should be fine. This is what it looks like with the above example ran.
![image](https://user-images.githubusercontent.com/25893271/114321653-8125f680-9ae1-11eb-8a40-8fbb9e40e469.png)

**Comment by Wolfr** [comment]
I am sorry but I can't seem to connect the two functions to return the right data, is this something you can PR? 🙏

**Comment by PaxPax** [comment]
Sure.
